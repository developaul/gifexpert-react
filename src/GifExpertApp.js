import React, { useState } from 'react';

import shortid from 'shortid';

import { AddCategory, GifGrid } from './components';

const GifExpertApp = () => {

	const [ categories, setCategory ] = useState([{ 
		nameCategory: 'One Punch',
		id: shortid.generate()
	}]);

	return (
		<>
			<h2>GifExpertApp</h2>
			
			<AddCategory
				categories={ categories }
				setCategory={ setCategory }
			/>

			<hr />

			<ol>
				{
					categories.map( category => (
						<GifGrid
							key={ category.id }
							{ ...category }		
						/>
					))
				}
			</ol>
		</>
	);
};

export default GifExpertApp;