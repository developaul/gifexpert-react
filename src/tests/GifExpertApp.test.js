import React from 'react';

import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import shortid from 'shortid';

import GifExpertApp from '../GifExpertApp';

describe( 'Pruebas en <GifExpertApp />', () => {
    
    test( 'Debe de renderizar correctamente el componente', () => {

        const wrapper = shallow( <GifExpertApp /> );

        expect( wrapper ).toMatchSnapshot();

    });

    test( 'Debe de mostrar una lista de categorias', () => {

        const categories = [{
    	  	nameCategory: 'One Punch',
    	  	id: shortid.generate()
        }]

        const wrapper = shallow( <GifExpertApp  /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'GifGrid' ).length ).toBe( categories.length );
    });

});