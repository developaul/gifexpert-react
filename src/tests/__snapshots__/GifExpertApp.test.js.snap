// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    categories={
      Array [
        Object {
          "id": "xHaoNgY3NP",
          "nameCategory": "One Punch",
        },
      ]
    }
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      id="xHaoNgY3NP"
      key="xHaoNgY3NP"
      nameCategory="One Punch"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifExpertApp /> Debe de renderizar correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    categories={
      Array [
        Object {
          "id": "reb3_TRR4",
          "nameCategory": "One Punch",
        },
      ]
    }
    setCategory={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      id="reb3_TRR4"
      key="reb3_TRR4"
      nameCategory="One Punch"
    />
  </ol>
</Fragment>
`;
